### PR TITLE
[mod] engines - add Onesearch engine

### PR DESCRIPTION
**Pull request by return42**
## What does this PR do?

Add Onesearch engine / Merged from @e-foundation's onesearch-engine branch [1] and slightly modfied / see [2].

[1] https://github.com/e-foundation/searx/tree/onesearch-engine
[2] https://github.com/searx/searx/pull/3065

**pro**: engine does fast response
~**hint**: engine is enabled by default~
**hint**: is literally just **Bing** --> could be a replacement for the bing engine / that is enabled by default but has some issues

**ToDo:** :

- [ ] Engine needs some polish .. this version does not support languages
- [ ] images: https://images.onesearch.com/yhs/search;?p=paris
- [ ] videos: https://video.onesearch.com/yhs/search;?p=paris
- [ ] news news: https://images.onesearch.com/yhs/search;?p=covid

-----
OneSearch is literally just **Bing** results flanked by ads that don’t track you despite being from a company that makes money tracking you [ref(https://lifehacker.com/is-yahoos-new-onesearch-engine-good-for-privacy-1841042875)].

According to the OneSearch privacy policy, search results will only be personalized based on location, which it will collect from IP addresses. OneSearch says that it will separate IP addresses from users and their search results [ref](https://www.theverge.com/2020/1/14/21065640/verizon-onesearch-privacy-tracking-yahoo-breach-hack).

## How to test this PR locally?

    !one search term

### CODE REVIEW

File: searx/data/engine_descriptions.json [hunk 1]
```diff
@@ -992,6 +1058,10 @@
   "brave":[
    "The Brave browser is a fast, private and secure web browser for PC, Mac and mobile. Download now to enjoy a faster ad-free browsing experience that saves data and battery life by blocking tracking software.",
    "https://brave.com/search/"
+  ],
+  "onesearch":[
+   "OneSearch is a free privacy-oriented search experience that lets you explore what you care about without unwanted profile or ad targeting. Take back control of your data today!",
+   "https://www.onesearch.com"
```
Comment (return42) [inline searx/data/engine_descriptions.json]: The script `./searxng_extra/update/update_engine_descriptions.py` has found a documentation / even there exists no wikidata entry.

Compare with https://github.com/searxng/searxng/pull/513/files#r751783403 / no words about Yahoo & Verizon, companies that are not known for protecting privacy.

File: searx/engines/onesearch.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+OneSearch is literally just Bing results flanked by ads that don’t track you
+despite being from a company that makes money tracking you [1].
+
+According to the OneSearch privacy policy, search results will only be
+personalized based on location, which it will collect from IP addresses.
+OneSearch says that it will separate IP addresses from users and their search
+results [2].
+
+[1] https://lifehacker.com/is-yahoos-new-onesearch-engine-good-for-privacy-1841042875
+[2] https://www.theverge.com/2020/1/14/21065640/verizon-onesearch-privacy-tracking-yahoo-breach-hack
```
Comment (return42) [inline searx/engines/onesearch.py:17]: compare with https://github.com/searxng/searxng/pull/513/files#r751782506

Comment (dalf) [reply searx/engines/onesearch.py:17]: I've added a wikidata entry: https://www.wikidata.org/wiki/Q109682354 so `update_engine_descriptions.py` should the wikidata description.

Comment (return42) [reply searx/engines/onesearch.py:17]: > I've added a wikidata entry

Very cool .. 

https://github.com/searxng/searxng/blob/c4a8c77026319185f516567f9b288157c2f87e20/searx/engines/onesearch.py#L28-L30

and I can confirm update_engine_descriptions.py works:

https://github.com/searxng/searxng/blob/356cff0692e0c59d263c17964dfb3a45ee3b2305/searx/data/engine_descriptions.json#L1053-L1056

https://github.com/searxng/searxng/blob/356cff0692e0c59d263c17964dfb3a45ee3b2305/searx/data/engine_descriptions.json#L1734-L1737

grafik

**Comment by dalf** [comment]
What is missing to make it ready for review?
The engine seems to work. Is there issue about the language selection?
Thank you for the engine description update (I realize the language identifier use underscore instead of hyphen in the master branch).

**Comment by return42** [comment]
> What is missing to make it ready for review?

OneSearch is literally just Bing results  --> we have bing with working language and paging support:

- https://github.com/searxng/searxng/pull/611

I think its worth to improve our bing engine. BTW: I don't know how I could say, but OneSearch _smells not good to me_ .. I expect problems with (one is already here https://github.com/searx/searx/pull/3099)

**Comment by dalf** [comment]
Ok let's close the PR for now.

**Comment by return42** [comment]
> Thank you for the engine description update 

FYI: I moved the update to https://github.com/searxng/searxng/pull/624

**Pull request closed without merging**